### PR TITLE
M47A2 Offensive Handgun

### PR DESCRIPTION
**Pull request by zc-byte-boxer**
# About the pull request

Adds the M47A2 offensive sidearm, a limited service 1911 variant used by limited USMC units, including FORECON. It finally adds a new, less ostentatious CO sidearm, and adds a unique sidearm for those on the backline.

# Explain why it's good for the game

Well, I concepted this pistol mainly as a new weapon for the commanding officer role that'd be more... less than the Mateba or Deagle. I'm very proud of the sprites I've got going with this, and since the lore fits, and it's a bit less powerful than the Mateba or Deagle, I'd like to include this as a non wl-exclusive item so more people may use it.

The intended design is to have a long-lasting sidearm that is sufficent for covering you in an emergency. It has damage inbetween the M44 and mod88, a slow RPM (help conserve that big magazine), low recoil, medium scatter, 14 rounds, and limited availability

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

https://github.com/user-attachments/assets/db1100bd-771a-4ce2-892a-d291331cd489
</details>

# Changelog

:cl: Byte-Boxer
add: Added the M47A2 Offensive Handgun. It's a mix of the 1911 and Hush Puppy pistol for usage by specialized marine units within the USCM. It spawns as the default pistol for IOs, is available to the CO, has a 20% FORECON spawn, and spawns in limited numbers (the same as the MOU) in req.
add: Added the M47A2 HP, AP, Impact, and regular magazines. Due to combat messages these use their own bullet, too. Although, the base shares its bullets with the 1911.
code: Added a new reload volume setting for coders with their guns. I added it so I could make the 1911 reload real load, s people hear its awesomeness.
code: Changed some backend code for IOs to be simpler.
soundadd: New sounds for the M47A2.
add: Fixes an error in logic wherein CO pistols were unacidable, but still could be blown up. Seeing as CO pistols are now the only way to BE, it seemed pertinent to fix this.
add: Added a falloff stat to heavy pistol rounds. Previously it had none.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -860,10 +861,11 @@ User can be passed as null, (a gun reloading itself for instance), so we need to
 		cock_gun(user)
 	user.visible_message(SPAN_NOTICE("[user] loads [magazine] into [src]!"),
 		SPAN_NOTICE("You load [magazine] into [src]!"), null, 3, CHAT_TYPE_COMBAT_ACTION)
-	if(reload_sound)
+	if((reload_sound && reload_volume))
```
Comment (harryob) [inline code/modules/projectiles/gun.dm:864]:
```suggestion
	if(reload_sound && reload_volume)
```

**Review by harryob — CHANGES_REQUESTED** [review]
otherwise seems fine

**Comment by zc-byte-boxer** [comment]
Assuming the workflows pass, could we get a TM for this so I can get the balance worked out for this? Sorry about being pushy.

@harryob @Drulikar

**Comment by zc-byte-boxer** [comment]
That should resolve the signal error.

File: code/modules/projectiles/guns/pistols.dm [hunk 2]
```diff
@@ -151,6 +151,56 @@
 /obj/item/weapon/gun/pistol/m1911/socom/equipped
 	starting_attachment_types = list(/obj/item/attachable/suppressor, /obj/item/attachable/lasersight, /obj/item/attachable/reflex)
 
+// A version of the M1911 for COs and FORECON.
```
Comment (Diegoflores31) [inline code/modules/projectiles/guns/pistols.dm:154]: Forecom quite literally already spawns with a m1911

File: code/modules/projectiles/guns/revolvers.dm [hunk 3]
```diff
@@ -548,7 +548,8 @@
 	icon = 'icons/obj/items/weapons/guns/guns_by_faction/uscm.dmi'
 	icon_state = "mateba"
 	item_state = "mateba"
-
+	unacidable = TRUE
```
Comment (Diegoflores31) [inline code/modules/projectiles/guns/revolvers.dm:551]: undocumented change

File: code/modules/projectiles/guns/pistols.dm [hunk 4]
```diff
@@ -236,6 +286,8 @@
 	icon = 'icons/obj/items/weapons/guns/guns_by_faction/colony.dmi'
 	icon_state = "c_deagle"
 	item_state = "c_deagle"
+	unacidable = TRUE
```
Comment (Diegoflores31) [inline code/modules/projectiles/guns/pistols.dm:289]: undocumented change

File: code/modules/projectiles/guns/pistols.dm [hunk 5]
```diff
@@ -151,6 +151,56 @@
...
+	accuracy_mult_unwielded = BASE_ACCURACY_MULT
+	scatter = SCATTER_AMOUNT_TIER_7
+	scatter_unwielded = SCATTER_AMOUNT_TIER_5
+	recoil =  RECOIL_AMOUNT_TIER_4
+	recoil_unwielded = RECOIL_AMOUNT_TIER_3
+	damage_mult = BASE_BULLET_DAMAGE_MULT + BULLET_DAMAGE_MULT_TIER_3
+
+/obj/item/weapon/gun/pistol/m47/co
+	name = "\improper custom M47A2 offensive handgun"
+	desc_lore = "The feeding ramp is polished to a mirror sheen, with a reinforced slide, and tightened locking mechanisms in the frame for added precision. The thumb safety is extended to make it easier on the finger, with a long-type trigger with non-slip grooves. A ring hammer. The base of the trigger guards's been filed down for a higher grip, too. And not only that, but nearly every other part of this gun has been expertly crafted and customized by USMC armourers."
```
Comment (Diegoflores31) [inline code/modules/projectiles/guns/pistols.dm:199]: USCM not USMC

File: code/datums/ammo/bullet/pistol.dm [hunk 6]
```diff
@@ -95,15 +95,41 @@
...
+	..()
+	RegisterSignal(src, COMSIG_AMMO_POINT_BLANK, PROC_REF(handle_battlefield_execution))
+
 /datum/ammo/bullet/pistol/heavy
 	name = "heavy pistol bullet"
 	headshot_state = HEADSHOT_OVERLAY_MEDIUM
 	accuracy = -HIT_ACCURACY_TIER_3
 	accuracy_var_low = PROJECTILE_VARIANCE_TIER_6
 	damage = 55
+	damage_falloff = DAMAGE_FALLOFF_TIER_5
```
Comment (Diegoflores31) [inline code/datums/ammo/bullet/pistol.dm:108]: undocumented ?

**Comment by Diegoflores31** [comment]
I personally dont think this weapon fits . and it is probably bloat 
rather VP78 could use some refreshment or if you really want to add a weapon you could add the M4A4 that Exists in the lore https://avp.fandom.com/wiki/M4A4_Pistol

File: code/modules/projectiles/guns/revolvers.dm (hunk 3 above)
Comment (zc-byte-boxer) [reply code/modules/projectiles/guns/revolvers.dm:551]: added. thanks for da reminder

File: code/modules/projectiles/guns/pistols.dm (hunk 5 above)
Comment (zc-byte-boxer) [reply code/modules/projectiles/guns/pistols.dm:199]: added. thanks for da reminder

File: code/modules/projectiles/guns/pistols.dm (hunk 4 above)
Comment (zc-byte-boxer) [reply code/modules/projectiles/guns/pistols.dm:289]: added. thanks for da reminder

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by Blundir — CHANGES_REQUESTED** [review]
needs some shading, details work, attachment position tweaked, make a new sprite for magazine or at least tweak current one, elaborated in dm

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by Blundir** [comment]
let me know when you'd return to it

**Pull request closed without merging**